packages/dds/map/src/directory.ts

```diff
@@ -1106,10 +1103,16 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 	private readonly pendingSubDirectories: Map<string, number[]> = new Map();
 
 	/**
-	 * Subdirectories that have been deleted locally but not yet ack'd from the server. This maintains the count
+	 * Subdirectories that have been deleted locally but not yet ack'd from the server. This maintains the record
 	 * of delete op that are pending or yet to be acked from server.
 	 */
-	private readonly pendingDeleteSubDirectoriesCount: Map<string, number> = new Map();
+	private readonly pendingDeleteSubDirectoriesTracker: Set<string> = new Set();
+
+	/**
+	 * Subdirectories that have been created locally but not yet ack'd from the server. This maintains the record
```

Comment:
could you clarify here and in the doc comments for `pendingSubDirectories` / `pendingKeys` how the two fields relate? It's not clear why we need both. It's probably also worth documenting that the `number[]` types in `pendingSubDirectories` and `pendingKeys` are lists of pendingMessageIds at which that key was modified.

Reply:
done